dhcp/src/lib.rs

```diff
@@ -273,6 +273,12 @@ impl<'a> Client<'a> {
         self.timeout = None;
     }
 
+    /// Get the IP address provided by DHCP.
+    #[inline]
+    pub fn ip(&self) -> Ipv4Addr {
+        self.ip
+    }
```

Comment:
This can be a little dangerous in that it can be called without a lease.

Maybe something like this instead?
```suggestion
    /// Get the leased IP address.
    pub fn leased_ip(&self) -> Option<Ipv4Addr> {
        if self.has_lease() {
            Some(self.ip)
        } else {
            None
        }
    }
```

Can you also add an entry to `dhcp/CHANGELOG.md` for this?  Something along the lines of

```markdown
- Added `Client.leased_ip` to return the currently leased IP address.
```